apps/src/components/download/step-summary.tsx

```diff
@@ -1,44 +1,70 @@
-	const analysisFields = climateVariable.getAnalysisFields?.() ?? [];
-	const analysisFieldValues = climateVariable.getAnalysisFieldValues?.() ?? {};
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Button } from '@/components/ui/button';
 
-	return (
-			<ul className="download-summary-bullet list-disc list-inside">
-				{climateVariable.getVersions().length > 0 && (<li key={version}><span className='text-dark-purple text-sm'>Version:</span> <span className="uppercase">{version || 'N/A'}</span></li>)}
-				{analysisFields.map(({ key, label }) => {
-					const value = analysisFieldValues[key] ?? '-';
-
-					return (
-						<li className="summary-item" key={key}>
-							<span className='text-gray-600 text-sm'>{__(label)}</span>: <span className="uppercase">{value}</span>
-						</li>
-					);
-				})}
-			</ul>
-	);
-};
-const StepSummary: React.FC = () => {
+// All <Step>Summary components
+import { StepSummaryVariableOptions } from '@/components/download/step-variable-options';
+import { StepSummaryLocation } from '@/components/download/step-location';
+import { StepSummaryAdditionalDetails } from '@/components/download/step-additional-details';
+import { StepSummarySendRequest } from '@/components/download/step-send-request';
+
+// Type-only imports for JSDoc @see references - these enable IDE navigation to step components
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepDataset } from '@/components/download/step-dataset';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepVariable } from '@/components/download/step-variable';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepVariableOptions } from '@/components/download/step-variable-options';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepLocation } from '@/components/download/step-location';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepAdditionalDetails } from '@/components/download/step-additional-details';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepSendRequest } from '@/components/download/step-send-request';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { default as StepResult } from '@/components/download/step-result';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { STEPS } from '@/components/download/config';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { DownloadProvider } from '@/context/download-provider';
+
+import type { StepSummaryData } from '@/types/download-form-interface';
+
+/**
+ * Summary on the side and visible throughout the download steps.
+ *
+ * This component displays a summary of user selections made in each step of the download form.
```

Comment:
This says it's a form, but I have issues finding HTML form and inputs. 

Which is a shame because most of this is about making sure we can't go to the next "step" and get to a previous step at the default state resetting the earlier steps we've jumped back.

And native HTML form from HTML 5 era (2009) features does this ([built-in form validation](https://developer.mozilla.org/en-US/docs/Learn_web_development/Extensions/Forms/Form_validation#using_built-in_form_validation )) with much less (so much less) code.